### PR TITLE
Update contributing guidelines, add a PR template

### PR DESCRIPTION
**Pull request by kale-stew**
This PR introduces a default template for all incoming PRs, so contributors have a better understanding of steps they need to adhere to to have their work merged.

It also updates the steps we've outlined in our Contributing doc based on my own experience contributing to this library earlier today. :-)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+-->
+
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
+
+## Checklist: (Feel free to delete this section upon completion)
+
+- [ ] All tests are passing
```
Comment (ryan-roemer) [inline .github/pull_request_template.md:14]: Let’s add a bullet for performance has not significantly decreased for the benchmark

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -106,24 +114,32 @@ $ yarn -s compress
...
+1. Confirm that all checks are passing:
+
+   ```sh
+   $ yarn run test
+   $ yarn run benchmark
+   ```
+
+2. Confirm we don't have any significant performance regressions (check out `master` for a baseline comparison on _your_ machine).
+
+3. Confirm you aren't impacting our benchmark stats.
```
Comment (ryan-roemer) [inline CONTRIBUTING.md:133]: This is really “bundle size” not “benchmark stats” (which instead deals with how many operations versus other libraries for performance). 

I think what we really want is a separated section for always checking the bundle size and updating the documents accordingly

Comment (kale-stew) [reply CONTRIBUTING.md:133]: We do have the Size section of this document, but it only outlines the steps necessary to check on that size, and not what to update in turn. Should I expand on the section in Size and keep the directions here? Or break out a different section entirely?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
-============
+# Contributing
 
 Thanks for contributing!
 
 ## Before you contribute
 
-This package is a fork of [fast-deep-equal](https://github.com/epoberezkin/fast-deep-equal). This library has added handling for React. Before contributing, _please make sure the issue relates directly to this library and not fast-deep-equals_.
+This package is a fork of [fast-deep-equal](https://github.com/epoberezkin/fast-deep-equal). This library has added handling for React.
+Before contributing, _please make sure the issue relates directly to this library and not fast-deep-equal's_.
```
Comment (chrisbolin) [inline CONTRIBUTING.md:8]: should just be fast-deep-equal, no **s**'s. My fault originally :)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
 
 We encourage pull requests concerning:
 
 * React features not handled in this library
-* Integrating updates from `fast-deep-equal`. This unfortunately, now requires more manual work to use the comment blocks in `index.js` to figure out what to paste and where.
-* Integrating tests from `fast-deep-equal`. This usually entails upgrading the `git`-based dependencies of `fast-deep-equal-git` and `npm`-published package of `fast-deep-equal` in `package.json:devDependencies`.
+* Integrating updates from `fast-deep-equal`
+
```
Comment (chrisbolin) [inline CONTRIBUTING.md:14]: is this just your prettier that adds the extra line breaks? i don't really care, i just want it to easily be consistent, and not a lot of dirty lines every time someone new changes the file.

Comment (chrisbolin) [reply CONTRIBUTING.md:14]: looks like these extra spaces actually make a visual difference:

image

Comment (chrisbolin) [reply CONTRIBUTING.md:14]: the 3 on the bottom, without extra line breaks, is what we want

**Pull request merged**